ingress/ingress.go

```diff
@@ -16,37 +16,69 @@ import (
 	"github.com/TykTechnologies/tyk-k8s/tyk"
 	"k8s.io/api/core/v1"
 	"k8s.io/api/extensions/v1beta1"
-	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/fields"
+	netv1beta1 "k8s.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-type Config struct{}
+type Config struct {
+	WatchNamespaces []string
+}
 
 var ctrl *ControlServer
 var log = logger.GetLogger("ingress")
 var opLog = sync.Map{}
+var runtimeScheme = runtime.NewScheme()
 
 const (
 	IngressAnnotation      = "kubernetes.io/ingress.class"
 	IngressAnnotationValue = "tyk"
+	defaultResync          = 2 * time.Minute
 )
 
 type ControlServer struct {
-	cfg               *Config
-	client            *kubernetes.Clientset
-	store             cache.Store
-	ingressController cache.Controller
-	podController     cache.Controller
-	stopCh            chan struct{}
+	cfg                 *Config
+	client              *kubernetes.Clientset
+	stopCh              chan struct{}
+	factories           map[string]informers.SharedInformerFactory
```

Comment:
I feel like calling the field `factories` is a bit non-idiomatic

Reply:
Well, it is what it is :man_shrugging: What would you suggest instead?

Reply:
Oh boy yeah lets leave it >.<